src/components/Page/Page.StickyActions.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react'
+import { StickyActionsUI } from './styles/Page.StickyActions.css'
+import { PageStickyActionsProps, PageStickyActionsState } from './Page.types'
+import { noop } from '../../utilities/other'
+
+class StickyActions extends React.PureComponent<
+  PageStickyActionsProps,
+  PageStickyActionsState
+> {
+  node: HTMLElement
+  observer: IntersectionObserver
+
+  static className = 'c-PageStickyActions'
+
+  static defaultProps = {
+    innerRef: noop,
+    offset: 10,
+    onStickyStart: noop,
+    onStickyEnd: noop,
+  }
+
+  state = {
+    isSticky: true,
+  }
+
+  componentDidMount() {
+    this.observerStart()
+  }
+
+  componentWillUnmount() {
+    this.observerStop()
+  }
+
+  observerStart() {
+    const { offset } = this.props
+    const observerOptions = {
+      root: null,
+      rootMargin: `0px 0px ${offset * -1}px 0px`,
+      threshold: 1.0,
+    }
+    this.observer = new IntersectionObserver(
```

Comment:
This is so awesome, I have had to implement this manually in the past and it wasn't that much fun